spec/parse_date/int_from_string_spec.rb

```diff
@@ -0,0 +1,578 @@
+    'Aug. 3rd, 1886' => '1886',
+    'Aug.st 4 1795' => '1795',
+    'Aug.t 16 1794' => '1794',
+    'Augt. 29, 1804' => '1804',
+    'August 1 1794' => '1794',
+
+    'Dec. 1 1792' => '1792',
+    'Dec.r 1 1792' => '1792',
+    'Dec.r 8th 1798' => '1798',
+    'Decb.r 1, 1789' => '1789',
+    'December 16 1795' => '1795',
+
+    'Feb 12 1800' => '1800',
+    'Feb. 10 1798' => '1798',
+    'Feb. 25, 1744]' => '1744',
+    'Feb.ry 12 1793' => '1793',
+    'Feb.ry 7th 1796' => '1796',
+    'Feb.y 1 1794' => '1794',
+    'Feb.y 13th 1798' => '1798',
+    'Feb.y 23rd 1799' => '1799',
+    '[Feb.y 18 1793]' => '1793',
+
+    'Jan. 1 1789' => '1789',
+    'Jan. 1. 1795' => '1795',
+    'Jan.y 15. 1795' => '1795',
+    'Jan.y 12st 1793' => '1793',
+    'Jan.y 18th 1790' => '1790',
+
+    'July 1 1796' => '1796',
+    'July 1. 1793' => '1793',
+    'July 13, 1787' => '1787',
+    'July 15th 1797' => '1797',
+
+    'June 1 1793' => '1793',
+    'June 1. 1800' => '1800',
+    'June1st.1805' => '1805',
+    'June 22, 1804' => '1804',
+    'July 23d 1792' => '1792',
+    'June 30th 1799' => '1799',
+    '[June 2 1793]' => '1793',
+
+    'May 9, 1795' => '1795',
+    'May 12 1792' => '1792',
+    'May 21st 1798' => '1798',
+    'May 15th 1798' => '1798',
+
+    'Mar. 1. 1792' => '1792',
+    'March 1 1795' => '1795',
+    'March 1.t 1797' => '1797',
+    'March 1, 1793' => '1793',
+    'March 1st 1797' => '1797',
+    'March 6th 1798' => '1798',
+    '[March 16 1798]' => '1798',
+
+    'Nov. 1. 1796' => '1796',
+    'Nov. 14th 1792' => '1792',
+    'Nov. 20 1789' => '1789',
+    'Nov.r 9, 1793' => '1793',
+    'Novem. 13th 1797' => '1797',
+    'Novembr 22nd 1794' => '1794',
+
+    'Oct 12 1792' => '1792',
+    'Oct 18th 1794' => '1794',
+    'Oct. 29 1796' => '1796',
+    'Oct. 11th 1794' => '1794',
+    'Oct.er 1st 1786' => '1786',
+    'Oct.r 25 1796' => '1796',
+    'Oct.r 25th 1794' => '1794',
+    'Octo.r 15 1795' => '1795',
+
+    'Sep.r 1, 1795' => '1795',
+    'Sep.tr 15.th 1796' => '1796',
+    'Sept.r 5th 1793' => '1793'
+  }
+  specific_day_ruby_parse_fail = {
+    # note ruby Date.parse only handles american or euro date order, not both ??
+    '1/30/1979' => '1979',
+    '10/20/1976' => '1976',
+    '5-18-2014' => '2014',
+    # year first
+    '1980-23-02' => '1980',
+    '1792 20 Dec' => '1792',
+    # text
+    'le 22 juin 1794' => '1794',
+    'mis au jour le 26 juillet 1791' => '1791',
+    'April 12 sd 1794' => '1794',
+    'Dec. 10 & 11, 1855' => '1855',
+    'January 22th [1800]' => '1800',
+    'June the 12, 1794' => '1794',
+    'Mai 1st 1789' => '1789',
+    'March 22 d. 1794' => '1794',
+    'N. 7 1796' => '1796',
+    'N[ovember] 21st 1786' => '1786',
+    'Oct. the 2.d 1793' => '1793'
+  }
+  # example string as key, expected parsed value as value
+  specific_day_2_digit_year = {
+    '1/2/79' => '1979',
+    '2/12/15' => '2015',
+    '6/11/99' => '1999',
+    '10/1/90' => '1990',
+    '10/21/08' => '2008',
```

Comment:
The Games of the IV Olympiad began 4/27/08 (1908).

Reply:
imperfect?  yes.  Same comment as above applies.   I think we're unlikely to see "4/27/08" as a string and have it refer to 1908.  I'm not sure when the slash syntax came into favor, but I suspect it is much more recent.